### DISCUSSION TITLE
watchReadContract vs watchContractEvent

### DISCUSSION THREAD

**Discussion by cilphex** [wevm/wagmi#1452] [Q&A]
I'm trying to listen for when a property on a contract changes, using wagmi.

As part of this I'm trying to understand the difference between `watchReadContract` and `watchContractEvent` in `wagmi/core`.

I think `watchContractEvent` is what I'm looking for, so long as the contract emits an event when the data that I'm interested in inside of the contract changes.

But then what is the purpose or use case of `watchReadContract`? Why would I want to have a callback fire when I read a contract? Is it so that I have a way to get a callback when not using `readContract` asynchronously?

**Comment by jxom** [comment]
Your assumptions are correct.

`watchContractEvent` = an action that listens for events on a contract
`watchReadContract` = an action that provides reactivity to `readContract` (this would be used if you wanted to use `wagmi` in Vanilla JS, Vue, etc and wanted `readContract` to update whenever the chain is changed)

**Comment by cilphex** [comment]
Would it be an appropriate pattern to use `readContract` once to load a piece of data, and then `watchContractEvent` to receive changes to that data through events, and ignore `watchReadContract`? Or is there a use case where I would additionally want `watchReadContract`?

Also, does `watchReadContract` do anything ever if I do not also call `readContract`?

And does the behavior of `watchReadContract` differ depending on whether I call `readContract` synchronously or asynchronously?
